src/zenml/cli/stack.py

```diff
@@ -174,6 +174,299 @@ def register_stack(
+    help="Include this to remove the model deployer from this stack.",
+    is_flag=True,
+    required=False,
+)
+def remove_stack_component(
+    stack_name: str,
+    container_registry_flag: Optional[bool] = False,
+    step_operator_flag: Optional[bool] = False,
+    secrets_manager_flag: Optional[bool] = False,
+    feature_store_flag: Optional[bool] = False,
+    model_deployer_flag: Optional[bool] = False,
+) -> None:
+    """Remove stack components from a stack."""
+    with console.status(f"Updating stack `{stack_name}`...\n"):
+        repo = Repository()
+        try:
+            current_stack = repo.get_stack(stack_name)
+        except KeyError:
+            cli_utils.error(
+                f"Stack `{stack_name}` cannot be updated as it does not exist.",
+            )
+        stack_components = current_stack.components
+
+        if container_registry_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.CONTAINER_REGISTRY
+            }
+
+        if step_operator_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.STEP_OPERATOR
+            }
+
+        if secrets_manager_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.SECRETS_MANAGER
+            }
+
+        if feature_store_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.FEATURE_STORE
+            }
+
+        if model_deployer_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.MODEL_DEPLOYER
+            }
+
+        stack_ = Stack.from_components(
+            name=stack_name, components=stack_components
+        )
+        repo.update_stack(stack_name, stack_)
+        cli_utils.declare(f"Stack `{stack_name}` successfully updated!")
+
+
+@stack.command("rename")
+@click.argument("current_stack_name", type=str, required=True)
+@click.argument("new_stack_name", type=str, required=True)
+def rename_stack(
+    current_stack_name: str,
+    new_stack_name: str,
+) -> None:
+    """Rename a stack."""
+    with console.status(f"Renaming stack `{current_stack_name}`...\n"):
+        repo = Repository()
+        try:
+            current_stack = repo.get_stack(current_stack_name)
+        except KeyError:
+            cli_utils.error(
+                f"Stack `{current_stack_name}` cannot be renamed as it does not exist.",
+            )
+        stack_components = current_stack.components
+
+        registered_stacks = [stack.name for stack in repo.stacks]
```

Comment:
```suggestion
        registered_stacks = {stack.name for stack in repo.stacks}
```
I would suggest using a set here as it makes conceptually more sense for a lookup